File: BookCatalogSystem/RubyApp/book.rb
# book.rb
# Represents a single book record in the catalog.

class Book
  attr_accessor :id, :title, :author, :genre, :publication_year, :quantity

  # Initialize a new Book instance
  def initialize(id, title, author, genre, publication_year, quantity = 1)
    @id = id
    @title = title
    @author = author
    @genre = genre
    @publication_year = publication_year
    @quantity = [quantity.to_i, 1].max  # Ensure at least 1 copy
  end

  # Human-readable string representation
  def to_s
    "ID: #{@id} | Title: #{@title} | Author: #{@author} | Genre: #{@genre} | Publication: #{@publication_year} | Stock: #{@quantity}"
  end
end

File: BookCatalogSystem/CsharpApp/Book.cs
public class Book
{
    public string Title { get; set; }
    public string Author { get; set; }
    public string Genre { get; set; }
    public int PublicationYear { get; set; }
    public int Quantity { get; set; }

    public Book(string title, string author, string genre, int year, int quantity = 1)
    {
        Title = title;
        Author = author;
        Genre = genre;
        PublicationYear = year;
        Quantity = quantity;
    }

    public override string ToString()
    {
        return $"{Title} | {Author} | {Genre} | {PublicationYear} | Stock: {Quantity}";
    }
}
